test/typescript/dispatch.ts

```diff
@@ -9,4 +9,4 @@ const dispatchResult: Action = dispatch({type: 'TYPE'});
 
 type Thunk<O> = () => O;
 
-const dispatchThunkResult: number = dispatch(() => 42);
+const dispatchThunkResult: number = dispatch<Thunk<number>, number>(() => 42);
```

Comment:
Could be written
```typescript
const dispatchThunkResult: number = dispatch((() => 42) as Thunk<number>);
```
which is shorter, more explicit and clear of what it does, provides exactly the same type safety and doesn't require people to put type casts all over their code if they doesn't want to.